Improved query string handling, tests

johnreutersward opened the pull request:
- Improved handling of query strings, only set fields will be encoded
- Added missing fields to streams, channels
- Added basic tests for streams, channels
- Made all endpoints that require authentication unexported for now

Commit:
Improved query string handling, tests

 - Improved handling of query strings, only set fields will be encoded
 - Added missing fields to streams, channels
 - Added basic tests for streams, channels
 - Made all endpoints that require authentication unexported for now

johnreutersward commented:
I realized that when ListOptions fields were not explicitly set they would of course get the default value which lead to some unintentional query strings. 

For example, all the bool fields would become false which means that the query contains '&Hls=false' which gives different results from Hls not being in the query at all. :smile:

mrshankly commented:
Maybe it would be better to use something like [this](https://github.com/google/go-querystring). It would make the code a bit cleaner since we can use `omitempty` to avoid missing fields appear in the query string, what do you think?

Also, regarding authentication, do you have any ideas about how it could work?

johnreutersward commented:
That looks good, I will look into it for sure.

As for auth I have no experience with oauth in Go but  https://godoc.org/code.google.com/p/goauth2/oauth looks promising.

Commit:
Added go-querystring library

Query strings are now specified in the ListOptions struct.
In each endpoint we simply need to encode them, a lot less code.
However, go-querystring does not encode bool as 'false' if the 'omitempty'
option is set, instead they are omitted. To get around this we can use
*bool without introducing any more types or changes in the code.
